### ISSUE TITLE
zpool state module needs support for disk vdev

### ISSUE THREAD

**Issue by aphor** [saltstack/salt#34762]
salt.modules.zpool does not support explicit (the second simplest example) "disk" type zpool vdev

Since the salt.states.zpool requires a vdev in layout kwarg in state.sls files, it is impossible to use this state module to create a simple single disk vdev zpool.

This should be a valid state
```
newpool:
  zpool.present:
    - config:
        import: false
        force: true
    - properties:
        comment: salty storage pool
    - layout:
        disk-0:
          /dev/disk0
          /dev/disk1
```
but produces an ambiguous error, assuming that the "disk" vdev is supposed to be interpreted by the salt.modules.zpool.create() as a device node on the filesystem
```
             if not os.path.exists(vdev):
                 ret[zpool][vdev] = 'not present on filesystem'
```

salt.states.zpool also does a similar (broken) check (DRY?) to validate the state layout.

**Comment by rallytime** [comment]
Thanks @aphor for the suggestion. Sounds good to me.

FYI @sjorge or @nmadhok - you guys might be interested in this feature request.

**Comment by sjorge** [comment]
@aphor if I understand correctly, you're looking for the equivalent of:
```zpool create tank /dev/disk0 /dev/disk1```

?

**Comment by aphor** [comment]
@sjorge yes, and I just realized I had botched my example syntax.

BTW: I have a PR on the way.

**Comment by aphor** [comment]
I think this is more about consistency in the YAML requirements to use the zpool state module.

#34770 <-- not too hairy.

**Comment by sjorge** [comment]
PR with a small docs update + support for the fake 'disk' vdev is going through my internal testing, will take a bit as I am on holiday on 3G internet.

**Comment by sjorge** [comment]
The fix seems to have made it into both 2016.3 and develop. @aphor is everything working 'as expected' now?

**Comment by aphor** [comment]
This is also working for me now. Thanks!
